src/test/java/io/candydoc/ddd/extract_ddd_concepts/ExtractDDDConceptsUseCaseTest.java

```diff
@@ -307,6 +317,250 @@ void find_interaction_between_two_different_concepts() throws IOException {
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            ConceptRuleViolated.builder()
+                .conceptName(
+                    "io.candydoc.sample.wrong_bounded_contexts.bounded_context.inner_shared_kernel")
+                .reason(
+                    "Shared kernel "
+                        + "shared_kernel_three"
+                        + " is not allowed in a bounded context.")
+                .build());
+  }
+
+  @Test
+  void shared_kernel_are_forbidden_inside_shared_kernel() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.wrong_bounded_contexts")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            ConceptRuleViolated.builder()
+                .conceptName(
+                    "io.candydoc.sample.wrong_bounded_contexts.shared_kernel.inner_shared_kernel")
+                .reason(
+                    "Shared kernel "
+                        + "shared_kernel_two"
+                        + " is not allowed in another shared kernel.")
+                .build());
+  }
+
+  @Test
+  void bounded_context_are_forbidden_inside_shared_kernel() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.wrong_bounded_contexts")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            ConceptRuleViolated.builder()
+                .conceptName(
+                    "io.candydoc.sample.wrong_bounded_contexts.shared_kernel.inner_bounded_context")
+                .reason(
+                    "Bounded context "
+                        + "bounded_context_three"
+                        + " is not allowed in a shared kernel.")
+                .build());
+  }
+
+  @ParameterizedTest
+  @MethodSource("forbidden_concepts_in_shared_kernel_examples")
+  void forbidden_concepts_in_shared_kernel(String conceptName, String ruleViolatedReason)
+      throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.wrong_bounded_contexts")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            ConceptRuleViolated.builder()
+                .conceptName(conceptName)
+                .reason(ruleViolatedReason)
+                .build());
+  }
+
+  public static Stream<Arguments> forbidden_concepts_in_shared_kernel_examples() {
```

Comment:
You can add shared kernel and bounded contexts in the arguments with their reason. If you do, you can remove the previous test "forbidden shared kernel, forbidden bounded context".

You can also make a ParameterizedTest for forbidden concepts in a Bounded Context